modules/evs/fsw/src/cfe_evs_utils.c

```diff
@@ -0,0 +1,624 @@
+        snprintf(PortMessage, CFE_EVS_MAX_PORT_MSG_LENGTH, "EVS Port2 %u/%u/%s %u: %s",
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.SpacecraftID,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.ProcessorID, EVS_PktPtr->Payload.PacketID.AppName,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.EventID, EVS_PktPtr->Payload.Message);
+        /* Send string event out port #2 */
+        EVS_OutputPort2(PortMessage);
+    }
+
+    if (((CFE_EVS_Global.EVS_TlmPkt.Payload.OutputPort & CFE_EVS_PORT3_BIT) >> 2) == true)
+    {
+        /* Copy event message to string format */
+        snprintf(PortMessage, CFE_EVS_MAX_PORT_MSG_LENGTH, "EVS Port3 %u/%u/%s %u: %s",
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.SpacecraftID,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.ProcessorID, EVS_PktPtr->Payload.PacketID.AppName,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.EventID, EVS_PktPtr->Payload.Message);
+        /* Send string event out port #3 */
+        EVS_OutputPort3(PortMessage);
+    }
+
+    if (((CFE_EVS_Global.EVS_TlmPkt.Payload.OutputPort & CFE_EVS_PORT4_BIT) >> 3) == true)
+    {
+        /* Copy event message to string format */
+        snprintf(PortMessage, CFE_EVS_MAX_PORT_MSG_LENGTH, "EVS Port4 %u/%u/%s %u: %s",
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.SpacecraftID,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.ProcessorID, EVS_PktPtr->Payload.PacketID.AppName,
+                 (unsigned int)EVS_PktPtr->Payload.PacketID.EventID, EVS_PktPtr->Payload.Message);
+        /* Send string event out port #4 */
+        EVS_OutputPort4(PortMessage);
+    }
+
+} /* End SendViaPorts */
+
+/*
+**             Function Prologue
+**
+** Function Name:      EVS_OutputPort1
+**
+** Purpose:  This routine sends the input message string out
+**           EVS port 1
+**
+** Assumptions and Notes:
+**
+*/
+void EVS_OutputPort1(char *Message)
+{
+    OS_printf("%s\n", Message);
+
+} /* End ES_OutputPort1 */
+
+/*
+**             Function Prologue
+**
+** Function Name:      EVS_OutputPort2
+**
+** Purpose:  This routine sends the input message string out
+**           EVS port 2
+**
+** Assumptions and Notes:
+**
+*/
+void EVS_OutputPort2(char *Message)
+{
+    OS_printf("%s\n", Message);
+
+} /* End ES_OutputPort2 */
+
+/*
+**             Function Prologue
+**
+** Function Name:      EVS_OutputPort3
+**
+** Purpose:  This routine sends the input message string out
+**           EVS port 3
+**
+** Assumptions and Notes:
+**
+*/
+void EVS_OutputPort3(char *Message)
+{
+    OS_printf("%s\n", Message);
+
+} /* End ES_OutputPort3 */
+
+/*
+**             Function Prologue
+**
+** Function Name:      EVS_OutputPort4
+**
+** Purpose:  This routine sends the input message string out
+**           EVS port 4
+**
+** Assumptions and Notes:
+**
+*/
+void EVS_OutputPort4(char *Message)
+{
+    OS_printf("%s\n", Message);
+
```

Comment:
Could these functions be combined into single function with Port as an input parameter?

Reply:
https://github.com/nasa/cFE/issues/94